server/routes/river-and-sea-levels.js

```diff
@@ -15,13 +15,34 @@ module.exports = [{
     let { location, riverIds, taCode, types } = getParameters(request)
     const referer = request.headers.referer
     let model, place, stations, targetArea
+    const rloiid = request.query['rloi-id']
+    const d = request.query.d
 
     // Convert river Ids into array
     riverIds = riverIds && riverIds.split(',')
 
     // Map types to db station types
     types = types && types.split(',')
 
+    if (rloiid) {
+      const id = parseInt(rloiid, 10)
+      const station = await floodService.getStationById(id, d)
```

Comment:
You can get the station information you need from the cached station data in floodservice so you don't have to go async and hit the database.
```
floodService.stationsGeojson.features.find(item => item.id === `stations.${rloiid}`)
```
That stationsGeojson is the object taken from geoserver every 15 minutes, so the geojson can be queried like a json object for the station data you need, the objects have "name" and also a geometry object to get the x and y from it